### DISCUSSION TITLE
Chunk Deletion Process

### DISCUSSION THREAD

**Discussion by xicxen** [Jikoo/Regionerator#191] [Q&A]
Hello,

I would like to know the process in which chunks are deleted (detailed if possible). I am working on adding functionality to the plugin where certain regions and chunks are restored from backed-up regions upon deletion. If there are no backups for a region or world, the plugin just functions as normal and deletes the regions/chunks. 

I have completed this for regions as a whole, but once it gets to managing specific chunks I haven't gotten it to work. I assume this is just because I don't know exactly how the plugin handles deleting specific chunks.

I know `AnvilRegion.write()` and `RegionFile.deleteChunk()` handle most of the deletion for regions and chunks.

(Just to aid in communication, I will understand code complexity as I am a fourth-year CS major, so don't be afraid to get into the nitty gritty!)

If it would be easier to talk elsewhere please let me know.

Thank you!

**Comment by Jikoo** [comment]
Don't have time to go into it right now, but Regionerator deletes chunks by wiping their pointers from the chunk header. What you're proposing would require a full region file write rather than editing the 4096 byte pointers and 4096 byte last edit timestamps, and would be a lot more expensive in terms of I/O. If you want basic details, I got most of my information for deleting chunks and the format in general from https://pastebin.com/niWTqLvk and can go into more specifics later

**Reply by Jikoo** [reply]
Okay, now that I have more time and have reread your question, I think the region file spec covers most everything you're asking about that I didn't cover myself. If you have any questions after you've read the .mcr spec (which is the same as .mca - the chunk NBT changed, not the region file format) let me know. It doesn't cover .mcc files as it's pretty dated, but otherwise it's largely still valid. I did write a bit about .mcc in [this post](https://github.com/Jikoo/Regionerator/issues/156#issuecomment-1448720001).

If you're pulling from a backup you could just use the existing chunk header, so you can theoretically copy verbatim every sector of the chunk to the earliest open sector set of appropriate size, write the new location in the region header, and, if the chunk is oversized, copy its `.mcc` file into the directory. You don't have to parse or interpret the raw chunk data in any way.
